src/Analyzers/CSharp/Analysis/OptimizeLinqMethodCallAnalysis.cs

```diff
@@ -690,6 +690,43 @@ public static void AnalyzeOrderByAndReverse(SyntaxNodeAnalysisContext context, i
         Report(context, invocationExpression, span, checkDirectives: true);
     }
 
+    // x.OrderBy(f => f) >>> x.Order()
+    public static void AnalyzeOrderByIdentity(SyntaxNodeAnalysisContext context, in SimpleMemberInvocationExpressionInfo invocationInfo)
+    {
+        InvocationExpressionSyntax invocationExpression = invocationInfo.InvocationExpression;
+
+        if (context.SemanticModel.GetSymbolInfo(invocationExpression).Symbol is IMethodSymbol methodSymbol)
+        {
+            INamedTypeSymbol containingType = methodSymbol.ContainingType;
+            IMethodSymbol orderMethod = containingType.GetMembers("Order")
+                .OfType<IMethodSymbol>()
+                .FirstOrDefault(m => m.Parameters.Length is 0);
```

Comment:
A number of parameters should be 1, it's an extension method. You can also use extension method `FindMember`:

```
IMethodSymbol orderMethod = containingType.FindMember<IMethodSymbol>("Order", m => m.Parameters.Length == 1)
```

Reply:
> A number of parameters should be 1, it's an extension method. You can also use extension method `FindMember`:
> 
> ```
> IMethodSymbol orderMethod = containingType.FindMember<IMethodSymbol>("Order", m => m.Parameters.Length == 1)
> ```

I couldn't figure out how exactly to call FindMember from `invocationInfo`.